MetovaTestKit/CellTesting/MTKCollectionViewTestCell.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  MTKTableViewTestCell.swift
+//  MetovaTestKit
+//
+//  Created by Nick Griffith on 2018-12-14
+//  Copyright © 2018 Metova. All rights reserved.
+//
+//  MIT License
+//
+//  Permission is hereby granted, free of charge, to any person obtaining
+//  a copy of this software and associated documentation files (the
+//  "Software"), to deal in the Software without restriction, including
+//  without limitation the rights to use, copy, modify, merge, publish,
+//  distribute, sublicense, and/or sell copies of the Software, and to
+//  permit persons to whom the Software is furnished to do so, subject to
+//  the following conditions:
+//
+//  The above copyright notice and this permission notice shall be
+//  included in all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+//  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+//  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+//  LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+//  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+//  WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import XCTest
+
+extension UICollectionView {
+    
+    /// This method attempts to produce a collection view cell from the collection view at the specified index path.  It then attempts to cast the cell to the specified type.  If either of these two fail, this function will produce a test failure and the test block will not be executed.  If both pass, the resulting cell, cast to the specified type, will be passed into the testBlock where the developer can run further assertions on the cell.
+    ///
+    /// - Parameters:
+    ///   - cellType: Type of cell to cast to before passing into test block.
+    ///   - indexPath: IndexPath to grab a cell from.
+    ///   - file: The file the test is called from.
+    ///   - line: The line the test is called from.
+    ///   - testBlock: A block of code containing tests to run.  This block is not guaranteed to execute.  If it does not execute, this method will fail the test.
+    /// - Throws: Rethrows any error thrown by the test block.  Does not throw any errors on its own.
+    func testCell<T: UICollectionViewCell>(
```

Comment:
So this is more of a question of discoverability more than anything. Do we have precedent within the MTK library for having an assertion/test method as an extension of a non-MTK specific class? In this case, I feel like it hinders discoverability of this method by burying it inside of `UICollectionView`. Thoughts?

Reply:
Hmm.  I could go either way.

On the one hand, this is kind of matching with the `MTKTestable` protocol, where we're calling `test` on objects that conform to the protocol.  On the other hand, in that case, the `test` function isn't designed to have test failures, where as this one definitely ease.

So, from the use case, the options are:

```swift
testVC.tableView.testCell(indexPath) { (cell: FooCell) in
    // assertions
}
```
--or--
```swift
MTKTestCell(in: tableView, at: indexPath) { cell: FooCell) in
    // assertions
}
```

Comparing the implementations side-by-side, any thoughts?

Moreover, with regards to discoverability, @lgauthier regularly contested that we should be dropping `MTK` prefix from all of the assertions.  Historically, I've argued for keeping them for discoverability and consistency with the default `XCTAssert` methods, which also keep their prefix (despite most other Swift things dropping the prefix).

For the foreseeable future, I expect to keep the `MTK` prefix, at least until Swift drops the `XCT` prefix from all of the default existing test assertions.  And that makes a fairly decent case for moving these methods to free-standing, rather than as UIKit extensions.

That leaves a couple questions though.

1. If we were to drop the `MTK` prefix throughout the whole library, in say a 3.0 release, does a method like `withCell(in: tableView, at: indexPath) { (cell: FooCell) in /* tests */ }` work better than `tableView.testCell(at: indexPath) { (cell: FooCell) in /* tests */ }`?

1. Do we actually even want to ever consider dropping the `MTK` prefix, even if the `XCT` is dropped from the default assertions?  I'd think _probably_ yes, but it means that in the case of a conflict on any of the assertions, we'd have to convert to `MetovaTestKit.withCell(//...`

1. What does this all mean for the existing `UIKit` assertions we have, actually?  Does this test being a `UITableView` extension suggest that, for example, `MTKAssertBarButtonItem(_, sends:, to: ) ` should be an extension of `UIBarButtonItem` instead of passing the bar button item in?  On this point, I'd argue that the case with `UITableView` and `UICollectionView` is different.

I can go either way, and am open to either implementation if you have a strong opinion here, just wanted to drop these things which should be considered in deciding the path here.

Reply:
For now, I'd prefer we stick to current naming conventions and make a PR at a later date to change them. I'd prefer the `MTKTestCell` at the moment. I feel discussing dropping the `MTK-` prefix is out of scope of this MR.